Replicator/Replicator.cc

```diff
@@ -980,4 +1100,78 @@ namespace litecore { namespace repl {
         request->respond(response);
     }
 
+
+    // Handles a "getCollections" request by looking up a peer checkpoint of each collection.
+    void Replicator::handleGetCollections(Retained<blip::MessageIn> request) {
+        auto root = request->JSONBody().asDict();
+        if (!root) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: no root"_sl});
+            return;
+        }
+        
+        auto checkpointIDs = root["checkpoint_ids"].asArray();
+        if (!checkpointIDs || checkpointIDs.empty()) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: no checkpoint_ids"_sl});
+            return;
+        }
+        
+        auto collections = root["collections"].asArray();
+        if (!collections || collections.empty()) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: no collections"_sl});
+            return;
+        }
+        
+        if (checkpointIDs.count() != collections.count()) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: mismatched checkpoint_ids and collections"_sl});
+            return;
+        }
+        
+        unordered_set<C4Database::CollectionSpec> specs;
+        for (auto &coll : _collections) {
+            specs.insert(coll->getSpec());
+        }
+        
+        MessageBuilder response(request);
+        auto &enc = response.jsonBody();
+        enc.beginArray();
+        
+        for (int i = 0; i < checkpointIDs.count(); i++) {
+            auto checkpointID = checkpointIDs[i].asString();
+            auto collectionPath = collections[i].asString();
+            
+            logInfo("Request to get peer checkpoint '%.*s' for collection '%.*s'",
+                    SPLAT(checkpointID), SPLAT(collectionPath));
+            
+            C4Database::CollectionSpec spec = Options::collectionPathToSpec(collectionPath);
+            if (specs.find(spec) == specs.end()) {
+                enc.writeNull();
+                continue;
+            }
+            
+            alloc_slice body, revID;
+            int status = 0;
+            try {
+                if (!Checkpointer::getPeerCheckpoint(_db->useLocked(), checkpointID, body, revID)) {
+                    enc.writeValue(Dict::emptyDict());
+                    continue;
+                }
+            } catch (...) {
+                C4Error::warnCurrentException("Replicator::handleGetCollections");
+                status = 502;
```

Comment:
Why 502 for unknown exception?

Reply:
It's the same error returned in GetRemoteCheckPoint(). I do not know the reason why it's 502 instead 400 or something else.